pkg/agent/proxy/types/types.go

```diff
@@ -85,3 +88,22 @@ func NewEndpointInfo(baseInfo *k8sproxy.BaseEndpointInfo, _ *k8sproxy.ServicePor
 	info := &EndpointInfo{BaseEndpointInfo: baseInfo}
 	return info
 }
+
+// GetLoadBalancerVIPInfos returns a slice of serviceStrings with the format
+// "ExternalIP:Port/Protocol" for LoadBalancer IPs
+func (s *ServiceInfo) GetLoadBalancerVIPStrings() []string {
+	return s.GenerateServiceStrings(s.LoadBalancerVIPs())
+}
+
+// GenerateServiceStrings generates service strings for the given ips in the format
+// "ExternalIP:Port/Protocol".
+func (s *ServiceInfo) GenerateServiceStrings(ips []net.IP) []string {
```

Comment:
My bad, maybe it is not a good practice to add such a method to `ServiceInfo`. The parameter `ips` can be any IP address list and it is a little weird. In my original thought, I think the function doesn't need an extra parameter, but we have delta updates for LoadBalancerIPs, we have to add the parameter.

How about removing the changes here and add a function to `proxier.go`:

```
func generateServiceInfoStrings(protocol binding.Protocol, port int, ips []net.IP) []string {
	var serviceStrs []string
	for _, ip := range ips {
		serviceStr := fmt.Sprintf("%s:%d/%s", ip, port, protocol)
		serviceStrs = append(serviceStrs, serviceStr)
	}
	return serviceStrs
}
```